### PR TITLE
2003 retrieve submit config

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do?

Fixes part of #2003
Enables submit, import, export and reset buttons in the configuration page

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [ ] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by manually running import, export, submit, reset, refreshing the page, changing tabs, importing malformed configs, etc.
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/monkey_island/cc/ui/src/components/configuration-components/ReformatHook.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export function reformatConfig(config, reverse = false) {
```
Comment (ilija-lazoroski) [inline monkey/monkey_island/cc/ui/src/components/configuration-components/ReformatHook.js:1]: :1st_place_medal:

File: monkey/monkey_island/cc/resources/agent_configuration.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def get(self):
     @jwt_required
     def post(self):
         try:
-            configuration_object = AgentConfigurationObject.from_json(request.data)
+            configuration_object = AgentConfigurationObject.from_mapping(request.json)
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/resources/agent_configuration.py:27]: If the request data type is json, flask automatically parse the data and puts it into request.json

Comment (mssalvatore) [reply monkey/monkey_island/cc/resources/agent_configuration.py:27]: What's the behavior if the request data type is not JSON and we try and access `request.json`?

**Review by mssalvatore — CHANGES_REQUESTED** [review]
eslint identified some issues. Check the TravicCI build for details.

File: monkey/monkey_island/cc/resources/agent_configuration.py (hunk 2 above)
Comment (VakarisZ) [reply monkey/monkey_island/cc/resources/agent_configuration.py:27]: 400, raise `BadRequestError`

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**